site/sfguides/src/intro_to_snowpark_container_services/intro_to_snowpark_container_services.md

```diff
@@ -0,0 +1,544 @@
+```
+You should see your `python-jupyter-snowpark:dev` image listed.
+
+### Configure and Push the Spec YAML
+Services in Snowpark Container Services are defined using YAML files. These YAML files configure all of the various parameters, etc. needed to run the containers within your Snowflake account. These YAMLs support a [large number of configurable parameter](), although we will not reference all of them here. Navigate to your local clone of `.../spcs-101-quickstart/src/jupyter-snowpark/jupyter-snowpark.yaml`, which should look like this:
+```yaml
+spec:
+  containers:
+    - name: jupyter-snowpark
+      image: <repository_hostname>/spcs_hol_db/public/image_repo/python-jupyter-snowpark:dev
+  endpoints:
+    - name: jupyter-snowpark
+      port: 8888
+      public: true
+  networkPolicyConfig:
+      allowInternetEgress: true
+
+```
+**Update the <repository_hostname> for your image** and save the file. Now that the spec file is updated, we need to push it to our Snowflake Stage so that we can reference it next in our `create service` statement. We will use snowcli to push the yaml file. From the terminal:
+```bash
+cd .../spcs-101-quickstart/src/jupyter-snowpark
+snow object stage copy ./jupyter-snowpark.yaml @specs --overwrite --connection spcs_hol
+```
+You can verify that your yaml was pushed successfully by running the following SQL using the Snowflake VSCode Extension or a SQL worksheet and verifying that the file is listed:
+```sql
+USE ROLE SPCS_USER_ROLE;
+LS @SPCS_HOL_DB.PUBLIC.SPECS;
+```
+
+### Create and Test the Service
+Once we have successfully pushed our image and our spec YAML, we have all of the components uploaded to Snowflake that we need in order to create our service. There are three components required to create the service: a service name, a compute pool the service will run on, and the spec file that defines the service. Run the following SQL to create our Jupyter service:
+```sql
+use role spcs_user_role;
+create service SPCS_HOL_DB.PUBLIC.jupyter_snowpark_service
+    in compute pool SPCS_HOL_POOL
+    from @specs
+    spec='jupyter-snowpark.yaml';
+```
+Run `CALL SYSTEM$GET_SERVICE_STATUS('SPCS_HOL_DB.PUBLIC.JUPYTER_SNOWPARK_SERVICE');` to verify that the service is successfully running. These commands are also spelled out in [`02_jupyter_service.sql`]().
+
+Since we specified that the `jupyter-snowpark` endpoint running on port `8888` would be `public: true` in our spec YAML, Snowflake is going to generate a URL for our service that can be used to access the service via our Web Browser. To get the URL, once the service is successfully in a `RUNNING` state, execute the following:
+```sql
+SHOW ENDPOINTS IN SERVICE JUPYTER_SNOWPARK_SERVICE;
+```
+Copy the `jupyter-snowpark` endpoint URL, and paste it in your browser. You will be asked to login to Snowflake via your username and password, after which you should successfully see your Jupyterlab instance running, all inside of Snowflake!
```

Comment:
We should give a note that the user that visits the endpoint will need the SPCS_USER_ROLE, and that their default role cannot be ACCOUNTADMIN, SECURITYADMIN, or ORGADMIN.